### ISSUE TITLE
GraphQL::Types::Relay::HasNodesField is not working

### ISSUE THREAD

**Issue by afurmanov** [rmosolgo/graphql-ruby#4049]
Generated GraphQL code does not add `node(id: ID!): Node, nodes(ids: [ID!]!): [Node]!` fields to the `query` root.

I've generated Rails project added a model and then run
`./bin/rails generate graphql:install --relay --batch --schema=EnterpriseSchema`  command.

**Versions**

- `graphql` version: 2.0.6 
- `rails` (or other framework): 7.0.1
- `graphql-batch` - included

**GraphQL schema**

This is what generator generates amongst other things:

```ruby
module Types
  class QueryType < Types::BaseObject
    # Add `node(id: ID!) and `nodes(ids: [ID!]!)`
    include GraphQL::Types::Relay::HasNodeField
    include GraphQL::Types::Relay::HasNodesField

    # Add root-level fields here.
    # They will be entry points for queries on your schema.

    # TODO: remove me
    field :test_field, String, null: false,
      description: "An example field added by the generator"
    def test_field
      "Hello World!"
    end
  end
end
```

When I inspect this schema in GraphiQL the `test_field` is visible, but `node` and `nodes` are not.
<img width="1005" alt="Screen Shot 2022-04-26 at 4 36 25 PM" src="https://user-images.githubusercontent.com/83098/165403961-e615fab3-d254-4400-bec3-80ca88927ee1.png">

I have the same issue with GraphQL 1.13.12 on the original project, however, for troubleshooting purposes, I've created a new, separate rails project from scratch, the latest and greatest code with almost zero customization (except adding `post` active record model using standard Rails generator). 

I expect to see `node` and `nodes` in GraphiQL, any ideas why they are not there?

Thanks!

**Comment by rmosolgo** [comment]
They aren't there because the `node` interface doesn't have any implementations. GraphQL-Ruby hides interfaces when they don't have any implementations, and it hides fields whose return types are hidden. This supports the visibility feature: https://graphql-ruby.org/authorization/visibility.html

This would be better handled if I actually implemented #2286. Duplicate-ish of #2674. 

To make the `node` and `nodes` fields show, make one of your object types implement the `Node` interface and make sure that object type is connected to the schema somehow (either as a return type of a field or using `orphan_types ...`). If that doesn't make the fields appear, please let me know!

**Comment by afurmanov** [comment]
Thanks, I've tried everything you recommended, without success: 

```
class EnterpriseSchema < GraphQL::Schema
  mutation(Types::MutationType)
  query(Types::QueryType)

  # GraphQL::Batch setup:
  use GraphQL::Batch
  orphan_types(Types::PostType)

  # GraphQL-Ruby calls this when something goes wrong while running a query:
  def self.type_error(err, context)
    # if err.is_a?(GraphQL::InvalidNullError)
    #   # report to your bug tracker here
    #   return nil
    # end
    super
  end

  # Union and Interface Resolution
  def self.resolve_type(abstract_type, obj, ctx)
    # TODO: Implement this method
    # to return the correct GraphQL object type for `obj`
    raise(GraphQL::RequiredImplementationMissingError)
  end

  # Relay-style Object Identification:

  # Return a string UUID for `object`
  def self.id_from_object(object, type_definition, query_ctx)
    # For example, use Rails' GlobalID library (https://github.com/rails/globalid):
    object_id = object.to_global_id.to_s
    # Remove this redundant prefix to make IDs shorter:
    object_id = object_id.sub("gid://#{GlobalID.app}/", "")
    encoded_id = Base64.urlsafe_encode64(object_id)
    # Remove the "=" padding
    encoded_id = encoded_id.sub(/=+/, "")
    # Add a type hint
    type_hint = type_definition.graphql_name.first
    "#{type_hint}_#{encoded_id}"
  end

  # Given a string UUID, find the object
  def self.object_from_id(encoded_id_with_hint, query_ctx)
    # For example, use Rails' GlobalID library (https://github.com/rails/globalid):
    # Split off the type hint
    _type_hint, encoded_id = encoded_id_with_hint.split("_", 2)
    # Decode the ID
    id = Base64.urlsafe_decode64(encoded_id)
    # Rebuild it for Rails then find the object:
    full_global_id = "gid://#{GlobalID.app}/#{id}"
    GlobalID::Locator.locate(full_global_id)
  end
end

module Types
  class QueryType < Types::BaseObject
    # Add `node(id: ID!) and `nodes(ids: [ID!]!)`
    include GraphQL::Types::Relay::HasNodeField
    include GraphQL::Types::Relay::HasNodesField

    # Add root-level fields here.
    # They will be entry points for queries on your schema.

    field :post, PostType

    def post
      Post.first
    end

    def self.visible?(context_)
      true
    end

    # TODO: remove me
    field :test_field, String, null: false,
      description: "An example field added by the generator"
    def test_field
      "Hello World!"
    end
  end
end

module Types
  class PostType < Types::BaseObject
    implements Types::NodeType
    field :id, ID, null: false
    field :title, String
    field :body, String
    field :published, Boolean
    field :created_at, GraphQL::Types::ISO8601DateTime, null: false
    field :updated_at, GraphQL::Types::ISO8601DateTime, null: false
    def self.visible?(context_)
      true
    end
  end
end
```

Post is shown in GraphiQL as implementing Node interface:
<img width="342" alt="Screen Shot 2022-04-28 at 5 10 25 PM" src="https://user-images.githubusercontent.com/83098/165862213-5483bf35-c4ed-4526-b52e-ab632f8f83bd.png">

The `post` field is visible as well:
<img width="346" alt="Screen Shot 2022-04-28 at 5 11 17 PM" src="https://user-images.githubusercontent.com/83098/165862260-fe1383b2-40e0-4f69-8292-f94d769f5c5e.png">

However, the query still does not have `node`, `nodes` fields.
